packages/components/src/color-palette/test/index.tsx

```diff
@@ -201,4 +212,41 @@ describe( 'ColorPalette', () => {
 			screen.getByRole( 'button', { name: 'Clear' } )
 		).toBeInTheDocument();
 	} );
+
+	it( 'should display the selected color name and value', async () => {
+		const user = userEvent.setup();
+
+		const { container } = render( <UnmockedColorPalette /> );
+
+		expect( screen.getByText( 'No color selected' ) ).toBeVisible();
+		expect(
+			// Relying on implementation details here, but it's the only way I could come up with.
+			// eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
+			container.querySelector(
+				'.components-color-palette__custom-color-value'
+			)
+		).toHaveTextContent( '' );
+
+		// Click the first unpressed button
+		await user.click(
+			screen.getAllByRole( 'button', {
+				name: /^Color:/,
+				pressed: false,
+			} )[ 0 ]
+		);
+
+		expect( screen.getByText( EXAMPLE_COLORS[ 0 ].name ) ).toBeVisible();
```

Comment:
Another set of checks that we could do when we select/unselect a color (although I'll leave it up to you if you think it's necessary) would be to check the label of the dropdown toggle, since it should say something like " The currently selected color is called ..." when there's indeed a selected color.

Reply:
Love that idea, and wish I'd thought of it myself!